bebraven/platform
# Fix routes for projects

### Description

This PR configures the routes so `Project`s use the `CustomContentsController`. The route automatically passes in the `type: "Project"` as a parameter.

This fixes the bug where the Content Editor won't load if you manually set `type: "Project"` to a `CustomContent` in our local development DB.

### Motivation and Context

The Content Editor only creates `CustomContent` (`custom_contents:type: nil`) objects currently . 

To create a `Project`, you need to use the development console (`devc`) and manually set `type: "Project"` on an existing `CustomContent` or a new `CustomContent`.

This is currently causing the `form_with` and `link_to` helpers to crash. See [this](https://guides.rubyonrails.org/form_helpers.html#relying-on-record-identification) for more details. 

### Test

- `platformweb/{course_contents, projects}/new` load

- **Save** works (note that `type: nil` here. Setting it to `"Project"` will be in a later PR)
<img width="1792" alt="Screen Shot 2020-10-26 at 3 44 29 PM" src="https://user-images.githubusercontent.com/62911934/97237135-d6748280-17a3-11eb-81e1-2ec2978607b4.png">

- `platformweb/custom_contents` renders **all** `CustomContent`s
<img width="1792" alt="Screen Shot 2020-10-26 at 3 56 58 PM" src="https://user-images.githubusercontent.com/62911934/97237171-e4c29e80-17a3-11eb-9296-2ce3b992a5e7.png">

- `platformweb/projects` renders **only** `Project`s
<img width="1792" alt="Screen Shot 2020-10-26 at 3 43 28 PM" src="https://user-images.githubusercontent.com/62911934/97237075-b218a600-17a3-11eb-9695-e3c530596fc5.png">

- check that `#show` renders for `type: nil` and `type: "Project"` (note that the latter has to be set in the dev console)

### Types of changes




### Checklist:
